File: php/features.php
<?php
declare(strict_types=1);

// StitchAiDesign SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class StitchAiDesignFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new StitchAiDesignBaseFeature();
            case "test":
                return new StitchAiDesignTestFeature();
            default:
                return new StitchAiDesignBaseFeature();
        }
    }
}

File: py/features.py
# StitchAiDesign SDK feature factory

from feature.base_feature import StitchAiDesignBaseFeature
from feature.test_feature import StitchAiDesignTestFeature


def _make_feature(name):
    features = {
        "base": lambda: StitchAiDesignBaseFeature(),
        "test": lambda: StitchAiDesignTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
